components/org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/ReceiptDAO.java

```diff
@@ -70,6 +70,16 @@
      */
     void deleteReceipt(String receiptID) throws ConsentManagementException;
 
+    /**
+     * Delete all {@link Receipt} of a given principal tenant id.
+     *
+     * @param tenantId Id of the tenant
+     * @throws ConsentManagementException
+     */
+    default void deleteReceiptsByPrincipalTenantId(int tenantId) throws ConsentManagementException {
```

Comment:
We might need to delete the receipts by service provide tenant ID as well. Please check

Reply:
Will check. THX!